app/controllers/tasks_controller.rb

```diff
@@ -17,6 +17,16 @@ def index
     @tasks = Task.all
   end
 
+  def complete
+    @task = Task.find(params[:id])
+    @task.mark_completed!
+    redirect_to tasks_path, notice: "Task completed successfully"
+  rescue ActiveRecord::RecordNotFound
+    redirect_to tasks_path, alert: "Task not found"
+  rescue StandardError => e
+    redirect_to tasks_path, alert: "Error completing task: #{e.message}"
```

Comment:
I don't like the idea of exposing raw errors to the users here - what are the rails conventions about "unknown errors" (AKA StandardError and descendants)